examples/miscellaneous/plot_pipeline_display.py

```diff
@@ -0,0 +1,218 @@
+# representation.
+
+from sklearn.compose import make_column_selector as selector
+from sklearn.preprocessing import OneHotEncoder, StandardScaler
+from sklearn.compose import ColumnTransformer
+from sklearn.linear_model import LogisticRegression
+from sklearn.pipeline import make_pipeline
+import pandas as pd
+from sklearn import set_config
+
+numerical_columns_selector = selector(dtype_exclude=object)
+categorical_columns_selector = selector(dtype_include=object)
+
+data = pd.DataFrame()
+
+numerical_columns = numerical_columns_selector(data)
+categorical_columns = categorical_columns_selector(data)
+
+categorical_preprocessor = OneHotEncoder(handle_unknown="ignore")
+numerical_preprocessor = StandardScaler()
+
+preprocessor = ColumnTransformer(
+    [
+        ("one-hot-encoder", categorical_preprocessor, categorical_columns),
+        ("standard-scaler", numerical_preprocessor, numerical_columns),
+    ]
+)
+
+pipe = make_pipeline(preprocessor, LogisticRegression(max_iter=500))
+
+# %%
+# To visualize the diagram, change to `display='diagram'`
+set_config(display="diagram")
+pipe  # click on the diagram below to see the details of each step
+
+# %%
+# Displaying a Grid Search over a Pipeline with a Classifier
+################################################################################
+# This section constructs a :class:`~sklearn.model_selection.GridSearchCV`
+# over a :class:`~sklearn.pipeline.Pipeline` with
+# :class:`~sklearn.ensemble.RandomForestClassifier` and displays its visual
+# representation.
+
+from sklearn.compose import make_column_selector as selector
+from sklearn.model_selection import GridSearchCV
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.impute import SimpleImputer
+from sklearn.preprocessing import OneHotEncoder, StandardScaler
+from sklearn.compose import ColumnTransformer
+from sklearn.pipeline import Pipeline
+from sklearn.pipeline import make_pipeline
+import numpy as np
+import pandas as pd
+from sklearn import set_config
+
+numerical_columns_selector = selector(dtype_exclude=object)
+categorical_columns_selector = selector(dtype_include=object)
+
+numeric_transformer = Pipeline(
+    steps=[
+        ("imputation_mean", SimpleImputer(missing_values=np.nan, strategy="mean")),
+        ("scaler", StandardScaler()),
+    ]
+)
+
+categorical_transformer = Pipeline(
+    steps=[
+        (
+            "imputation_constant",
+            SimpleImputer(fill_value="missing", strategy="constant"),
+        ),
+        ("onehot", OneHotEncoder(handle_unknown="ignore")),
+    ]
+)
+
+data = pd.DataFrame()
+
+numerical_columns = numerical_columns_selector(data)
+categorical_columns = categorical_columns_selector(data)
+
+preprocessor = ColumnTransformer(
+    transformers=[
+        ("num", numeric_transformer, numerical_columns),
+        ("cat", categorical_transformer, categorical_columns),
+    ]
+)
+
+pipe = Pipeline(
```

Comment:
For this version, I think it would be good to have a simplier pipeline and show case what happens with gridsearch.

```python
preprocessor = ColumnTransformer(
    [
        ("categorical", OneHotEncoder() ["state"]),
        ("numerical", StandardScalar(), ["area", "population"]),
    ]
)

pipe = Pipeline(steps=[("preprocessor", preprocessor), ("classifier", RandomForestClassifier())])
```